mantidproject/mantid
# Automatically choose a contrasting axes color

Description of work.

This adds an option to automatically choose the color of the orientation axes, axes grid, and color scale text so that it will contrast with the background color. This makes it quick to go between a white (for printing) and gray (for viewing on the screen) background.

**To test:**



1. Open the VSI preferences and ensure that "Automatic axes color selection" is checked. If not, enable it.
2. Load a MDWorkspace in the VSI and set the background color to black. Verify that the orientation axes, axes grid, and color scale text are now white.
3. Set the background color to white. Verify that the orientation axes, axes grid, and color scale text are now black.
4. Open the VSI preferences and uncheck "Automatic axes color selection". Restart the VSI.
5. Load a MDWorkspace in the VSI and verify that the axes colors don't vary with the background color.

Fixes #15696.

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
